content/doc/tutorials/build-a-dotnet-web-app-with-jenkins.adoc

```diff
@@ -0,0 +1,363 @@
++
+[source,groovy]
+----
+        stage('Test') {
+            steps {
+                sh 'dotnet test --no-build --no-restore --collect "XPlat Code Coverage"'
+            }
+            post {
+                always {
+                    recordCoverage(tools: [[parser: 'COBERTURA', pattern: '**/*.xml']], sourceDirectories: [[path: 'SimpleWebApi.Test/TestResults']])
+                }
+            }
+        }
+----
+so that you end up with:
++
+[source,groovy]
+----
+pipeline {
+    agent any
+    stages {
+        stage('Build') {
+            steps {
+                sh 'dotnet restore'
+                sh 'dotnet build --no-restore'
+            }
+        }
+        stage('Test') { // <1>
+            steps {
+                sh 'dotnet test --no-build --no-restore --collect "XPlat Code Coverage"' // <2>
+            }
+            post {
+                always {
+                    recordCoverage(tools: [[parser: 'COBERTURA', pattern: '**/*.xml']], sourceDirectories: [[path: 'SimpleWebApi.Test/TestResults']])  // <3>
+                }
+            }
+        }
+    }
+}
+----
+<1> Defines a link:/doc/book/pipeline/syntax/#stage[`stage`] (directive) called `Test` that appears on the Jenkins UI.
+<2> This link:/doc/pipeline/steps/workflow-durable-task-step/#code-sh-code-shell-script[`sh`] step executes the `dotnet test` command to run the unit/integration test on your .NET web app.
+This command also generates a Cobertura XML report, which is saved to the `SimpleWebApi.Test/TestResults` directory within the `/home/jenkins/agent/workspace/dotnet-tutorial` directory in the Jenkins agent container.
+<3> This link:/doc/pipeline/steps/coverage[`recordCoverage`] step (provided by the link:/doc/pipeline/steps/coverage[Coverage Plugin]), archives the Cobertura XML report generated by the `dotnet test` command above, and displays the results through the Jenkins interface.
+The link:/doc/book/pipeline/syntax/#post[`post`] section's `always` condition that contains this `recordCoverage` step ensures that the step is _always_ executed _at the completion_ of the `Test` stage, regardless of the stage's outcome.
+. Save your edited `Jenkinsfile` and commit it to your local `simple-dotnet-web-app` Git repository.
+Within the `simple-dotnet-web-app` directory, run the commands:
++
+.. `git stage .`
+.. `git commit -m "Add 'Test' stage"`
+.. `git push` to push your changes to your forked repository on GitHub, so it can be picked up by Jenkins.
+
+. In Jenkins, go back to *Dashboard* if necessary, then *dotnet-tutorial* and launch another build using *Build Now*.
+. After a while, a new column titled *Test* appears in the *Stage View*.
+
+[.boxshadow]
+image:tutorials/dotnet/test-stage-added.png[alt="Test stage runs successfully",width=100%]
+
+You can select *#2* (or the number representing your last build) in the *Builds* widget to see the details of the build.
```

Comment:
```suggestion
You can select *#2* (or the number representing your last build) in the *Stage View* or *Builds* widget to see the details of the build.
```
Suggestion to be explicit where the links can be accessed. Also, would it be possible to update this screenshot to include the Builds widget with build 2?